java/ql/test/utils/model-generator/typebasedflow/p/TypeBasedComplex.java

```diff
@@ -0,0 +1,57 @@
+package p;
+
+import java.util.List;
+import java.util.function.Function;
+
+public class TypeBasedComplex<T> {
+
+    public void addMany(List<T> xs) {
+    }
+
+    public List<T> getMany() {
+        return null;
+    }
+
+    public Integer apply(Function<T, Integer> f) {
+        return null;
+    }
+
+    public <T1, T2> T2 apply2(T1 x, Function<T1, T2> f) {
```

Comment:
This type signature doesn't refer to `T`. Was that intentional or was the method perhaps meant to be static?

Reply:
If it's meant as-is, then it deserves an explanatory comment.

Reply:
Yes, that as intentional. The test is there to check that no summary that accesses the *implicit* field corresponding to the field of type `T` will be generated (even for instance methods).

Reply:
Comment added.